src/components/common/rest_api.ts

```diff
@@ -115,3 +115,23 @@ function toFormData(file: File) {
   formData.append('upload', file);
   return formData;
 }
+
+export function isCloudWorkstation(url: string) {
+  return url.includes('.cloudworkstations.dev');
+}
+
+// This is a very minimal wrapper around fetch that lets us more easily make
+// broad changes to API calls.
+// Previously, we had a ton a raw fetch calls throughout the codebase, so any broad changes needed to be made all over the place.
+export async function fetchMaybeWithCredentials(
+  url: string,
+  opts?: RequestInit
+): Promise<Response> {
+  if (!opts) {
+    opts = {};
+  }
+  if (isCloudWorkstation(url)) {
+    opts.credentials = 'include';
```

Comment:
I think you can just do

```suggestion
    opts = {...opts, credentials: 'include'};
```
It will handle undefined without Line 130-132 and it avoids modifying the argument